plugin.py

```diff
@@ -795,10 +797,19 @@ def list_items(self) -> List[sublime.ListInputItem]:
             if os.path.isdir(os.path.join(julia_environments_path, env))
         ]
         env_paths = [os.path.join(julia_environments_path, env) for env in env_names]
-        # Add workspace folders if they are valid Julia project environments
+        # Add all environments withing a working folder.
         for folder in self.workspace_folders:
-            if folder.path not in env_paths and is_julia_environment(folder.path):
-                items.append(sublime.ListInputItem(folder.name, folder.path, kind=KIND_WORKSPACE_FOLDER))
+            for subdir, dirs, files in os.walk(folder.path):
+                folderpath = os.path.join(folder.path, subdir)
+                if folderpath not in env_paths and is_julia_environment(folderpath):
+                    if os.path.relpath(subdir, folder.path) == '.':
+                        env_name = os.path.join(os.path.basename(folder.path))
```

Comment:
Here `os.path.join` is superfluous with a single argument. And it should be possible to simplify this a bit by extracting the `os.path.basename` and `os.path.relpath` parst into variables.

Probably something like
```python
basedir = os.path.basename(folder.path)
relpath = os.path.relpath(subdir, folder.path)
env_name = basedir if relpath == '.' else os.path.join(basedir, relpath)
```